### PR TITLE
Make API call to DTP to get rule doc URL

### PR DESCRIPTION
**Pull request by dtang-parasoft**

### CODE REVIEW

**Comment by dtang-parasoft** [comment]
Need to integrate [Alfie` s](https://github.com/parasoft/parasoft-findings-vsts/pull/14) code first

**Comment by yshi-parasoft** [comment]
Have you tried using the "https" from standard library?
https://nodejs.org/api/https.html

File: PublishParasoftTestResults/publishtestresults.ts [hunk 1]
```diff
@@ -463,3 +477,70 @@ function isNullOrWhitespace(input: any) {
...
+
+function getRuleDoc(ruleId: string, analyzerId: string, apiVersion: number): Promise<any> {
+    const credentials = Buffer.from(`${dtpUsername}:${dtpPassword}`).toString('base64');
+    let url = dtpBaseUrl+ "grs/api/v" + apiVersion +"/rules/doc?rule=" + ruleId + "&analyzerId=" + analyzerId;
+    return axios.default.get(url, {
+        headers: {
+            'Authorization': `Basic ${credentials}`
+        },
+        httpsAgent: new https.Agent({
+            rejectUnauthorized: false
```
Comment (yshi-parasoft) [inline PublishParasoftTestResults/publishtestresults.ts:543]: Disabling the SSL/TLS certificate verification may not be good. We can create separate task to figure out how to handle this better.

File: PublishParasoftTestResults/publishtestresults.ts [hunk 2]
```diff
@@ -463,3 +477,70 @@ function isNullOrWhitespace(input: any) {
...
+    }
+    return Promise.resolve();
+}
+
+function doGetRulesDocs(index: number): Promise<any> {
+    const ruleId = Array.from(ruleAnalyzerPairs.keys())[index];
+    // AnalyzerId cannot be undefined.
+    const analyzerId = ruleAnalyzerPairs.get(ruleId) || "";
+
+    return getRuleDoc(ruleId, analyzerId, 1.6)
```
[the remark this answers is not in the document]
Comment (rwang-parasoft) [reply PublishParasoftTestResults/publishtestresults.ts:501]: The DTP wich doesn't support V1.6 API is considered to be a legacy DTP. In Xtest use V1.6 and V1.0 API to get dtp services. So we use  V1.6 API to get docsUrl first, if we can't get the expected result wo use V1.0 API to get docsUrl  again on line [512](https://github.com/parasoft/parasoft-findings-vsts/pull/17/files/e2728f63fe8a017bc776f3d69fba98050b218b0c#diff-e2147b98e067deb51fe7f4fd39a26d5ded50b2208169d78f2627b9b3d69b3b2bR512).

**Pull request closed without merging**